### PR TITLE
fix(curriculum): regex to target html elements using jQuery to captur…

### LINKED ISSUE #38648: jQuery: Target HTML Elements With Selectors Using jQuery
- https://www.freecodecamp.org/learn/front-end-libraries/jquery/target-html-elements-with-selectors-using-jquery

Code:
```
$(document).ready(function()
{
  $("button").addClass("animated bounce");
});
```
Errors: `You should use the jQuery addClass() function to give the classes animated and bounce to your button elements.
Your jQuery code should be within the $(document).ready(); function.`

Code:
```
$(document).ready(function(){
  $("button").addClass("animated bounce");
});
```
Error: `You should use the jQuery addClass() function to give the classes animated and bounce to your button elements.`

But:
```
$(document).ready(function() {
  $("button").addClass("animated bounce");
});
```
passes flawlessly. The test regular expressions are obviously poorly written.

### PR DESCRIPTION
**Pull request by AshNaz87**
…e white spaces

Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `master` branch of freeCodeCamp.
- [x] All the files I changed are in the same world language, for example: only English changes, or only Chinese changes, etc.

Closes #38648

### CODE REVIEW

**Review by moT01 — CHANGES_REQUESTED** [review]
I think this could be changed to use jQuery instead of regex. Spaces can be between almost anything in javascript...

```
$
(
document
)
.
ready
(
function
...
```
That would be functional, But kind of a pain to check for all those spaces/new lines.

I checked, and the classes only get applied if the code is in the `document.ready` function.
So I would change this to check the classes of the buttons somehow. e.g. `assert($('buttons').classList.includes('animated' && 'bounce'))`
That syntax is wrong, but something along those lines is how we should do it I think.

**Comment by nhcarrigan** [comment]
@AshNaz87 Can you provide an update on the implementation of the requested changes?

**Comment by AshNaz87** [comment]
Hi @nhcarrigan 

Please feel free to edit

File: curriculum/challenges/english/03-front-end-libraries/jquery/target-html-elements-with-selectors-using-jquery.english.md [hunk 1]
```diff
@@ -32,8 +32,7 @@ tests:
   - text: You should only use jQuery to add these classes to the element.
     testString: 'assert(!code.match(/class.*animated/g));'
   - text: Your jQuery code should be within the <code>$(document).ready();</code> function.
```
Comment (Sky020) [inline curriculum/challenges/english/03-front-end-libraries/jquery/target-html-elements-with-selectors-using-jquery.english.md:34]:
```suggestion
```

File: curriculum/challenges/english/03-front-end-libraries/jquery/target-html-elements-with-selectors-using-jquery.english.md [hunk 2]
```diff
@@ -32,8 +32,7 @@ tests:
...
-    testString: assert(code.match(/\$\(document\)\.ready\(function.*(\s|\n)*.*button.*.addClass.*\);/g));
-
+    testString: assert(code.match(/\$\(document\)\.ready\(function\s*\(\)\s*{\n\s*\$.*button.*.addClass.*/g));
```
Comment (Sky020) [inline curriculum/challenges/english/03-front-end-libraries/jquery/target-html-elements-with-selectors-using-jquery.english.md:35]:
```suggestion
```

**Review by ShaunSHamilton — CHANGES_REQUESTED** [review]
@moT01 , Seeing as the first test is already checking for the classes added to the button, can we just remove this test? It is redundant, and error prone.

File: curriculum/challenges/english/03-front-end-libraries/jquery/target-html-elements-with-selectors-using-jquery.english.md (hunk 2 above)
Comment (RandellDawson) [reply curriculum/challenges/english/03-front-end-libraries/jquery/target-html-elements-with-selectors-using-jquery.english.md:35]: Not sure how it is redundant.  Based on the instructions, we expect the user to put the other code inside the `$(document).ready() function`.  The first test only checks if they are using the the `addClass` method regardless of where it is placed.

Comment (Sky020) [reply curriculum/challenges/english/03-front-end-libraries/jquery/target-html-elements-with-selectors-using-jquery.english.md:35]: The first test is:
```
assert($("button").hasClass("animated") && $("button").hasClass("bounce"));
```
Does this not use jQuery to test if the class has been successfully added to the `button` element?

Comment (RandellDawson) [reply curriculum/challenges/english/03-front-end-libraries/jquery/target-html-elements-with-selectors-using-jquery.english.md:35]: @Sky020 Yes, but it does not validate that the `hasClass` method code is inside the `$(document).ready()` function, which is part of the instructions.

Comment (Sky020) [reply curriculum/challenges/english/03-front-end-libraries/jquery/target-html-elements-with-selectors-using-jquery.english.md:35]: The user is expected to use the `addClass` method to add two classes to the `button` element. So, if the `button` has those two classes added to it, either the user has used DOM methods to add the class (which requires adding a script tag to the bottom of the editor with the code, and they would need to do something similar to:
```
<script>
  document.querySelectorAll('button').forEach(ele => {
    ele.classList
    .add('animated');
    // The random spacing is necessary to fool the second test
    ele.classList
    .add('bounce');
  });
</script>
```
, **or** they have placed the `addClass` code within the `document.ready` function. If the `addClass` method is not within the `document.ready` function, the first test will not pass. Unless code like this has been written:
```
  $(document).ready(function() {
    callback()
  });
  function callback() {
    $('button').addClass("animated bounce");
  }
```

I think fooling this test would be an acceptable way to pass. That is why I think it is redundant.

Comment (moT01) [inline curriculum/challenges/english/03-front-end-libraries/jquery/target-html-elements-with-selectors-using-jquery.english.md:35]:
```suggestion
    testString: 'assert($("button").hasClass("animated") && $("button").hasClass("bounce"));'
```

I vote to just use the same test as the first test. Like @Sky020 mentioned - there's ways around it, but it pretty closely ensures that their code is in the function. And I like keeping the test text there as a reminder for users to put their code in it.

Comment (moT01) [inline curriculum/challenges/english/03-front-end-libraries/jquery/target-html-elements-with-selectors-using-jquery.english.md:35]:
```suggestion
    testString: assert(code.match(/\$\(document\)\.ready\(function\s*\(\)\s*{\s*\$\(\s*('|")button\1\s*\)\s*\.\s*addClass[\s\S]*\)/g));
```

Comment (moT01) [reply curriculum/challenges/english/03-front-end-libraries/jquery/target-html-elements-with-selectors-using-jquery.english.md:35]: @RandellDawson and I were thinking we should go back to just using regex. The thinking is that users would have to have the correct code in order to pass even though their code may be in the correct area.

Comment (RandellDawson) [reply curriculum/challenges/english/03-front-end-libraries/jquery/target-html-elements-with-selectors-using-jquery.english.md:35]: @Sky020 If you are ok with @moT01's suggestion above, I am fine with it, however, we could go with a shorter version by just removing the spaces:
```js
assert(code.replace(/\s/g, '').match(/\$\(document\)\.ready\(function\(\)\{\$/g));
```
If you are fine with this, feel free to either commit @moT01's or my suggestion above. Then I can approve/merge this PR.  Thanks.

Comment (Sky020) [inline curriculum/challenges/english/03-front-end-libraries/jquery/target-html-elements-with-selectors-using-jquery.english.md:35]:
```suggestion
    testString: "assert(code.replace(/\s/g, '').match(/\$\(document\)\.ready\(function\(\)\{\$/g));"
```

**Review by ShaunSHamilton — DISMISSED** [review]
@RandellDawson LGTM 👍 
I am happy committing your addition on top of Tom's suggestion

Edit: I need to sort out the linting error.

**Comment by RandellDawson** [comment]
@Sky020 The linter does not like my suggestion.

File: curriculum/challenges/english/03-front-end-libraries/jquery/target-html-elements-with-selectors-using-jquery.english.md [hunk 3]
```diff
@@ -32,7 +32,7 @@ tests:
   - text: You should only use jQuery to add these classes to the element.
     testString: 'assert(!code.match(/class.*animated/g));'
   - text: Your jQuery code should be within the <code>$(document).ready();</code> function.
-    testString: assert(code.match(/\$\(document\)\.ready\(function\s*\(\)\s*{\n\s*\$.*button.*.addClass.*/g));
+    testString: "assert(code.replace(/\s/g, '').match(/\$\(document\)\.ready\(function\(\)\{\$/g));"
```
Comment (Sky020) [inline curriculum/challenges/english/03-front-end-libraries/jquery/target-html-elements-with-selectors-using-jquery.english.md:35]:
```suggestion
    testString: 'assert(code.replace(/\s/g, '''').match(/\$\(document\)\.ready\(function\(\)\{\$/g));'
```

**Review by ShaunSHamilton — APPROVED** [review]
Sorry about that, Randell. All appears fine, now.

I am not sure why the linter was unhappy with the double quotes.

**Review by moT01 — APPROVED** [review]
LGTM @AshNaz87 🎉 Thanks for contributing to freeCodeCamp 🎉

**Pull request merged**